console-subscriber/tests/support/task.rs

```diff
@@ -0,0 +1,228 @@
+use std::{error, fmt};
+
+use super::MAIN_TASK_NAME;
+
+/// An actual task
+///
+/// This struct contains the values recorded from the console subscriber
+/// client and represents what is known about an actual task running on
+/// the test's runtime.
+#[derive(Clone, Debug)]
+pub(super) struct ActualTask {
+    pub(super) id: u64,
+    pub(super) name: Option<String>,
+    pub(super) wakes: u64,
+    pub(super) self_wakes: u64,
+}
+
+impl ActualTask {
+    pub(super) fn new(id: u64) -> Self {
+        Self {
+            id,
+            name: None,
+            wakes: 0,
+            self_wakes: 0,
+        }
+    }
+}
+
+/// An error in task validation.
+pub(super) struct TaskValidationFailure {
+    /// The expected task whose expectations were not met.
+    expected: ExpectedTask,
+    /// The actual task which failed the validation
+    actual: Option<ActualTask>,
+    /// A textual description of the validation failure
+    failure: String,
+}
+
+impl error::Error for TaskValidationFailure {}
+
+impl fmt::Display for TaskValidationFailure {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.failure)
+    }
+}
+
+impl fmt::Debug for TaskValidationFailure {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &self.actual {
+            Some(actual) => write!(
+                f,
+                "Task Validation Failed!\n  Expected Task: {expected:?}\n  Actual Task:   {actual:?}\n  Failure:       {failure}",
+                expected = self.expected, failure = self.failure),
```

Comment:
nit: weird formatting
```suggestion
                expected = self.expected,
                failure = self.failure,
            ),
```